### PR TITLE
Feature/dockstore layer only

### PR DESCRIPTION
**Pull request by keiranmraine**
Minimise the places we have installation code, will be more difficult for wxs/wgs container but this was trivial.

### CODE REVIEW

**Review by byb121 — CHANGES_REQUESTED** [review]

File: scripts/ds-cgpmap.pl [hunk 1]
```diff
@@ -179,14 +199,13 @@ =head1 OPTION DETAILS
...
 
-=item B<-scramble>
+=item B<-seqslice>
 
-Single quoted string of parameters to pass to Scramble when '-c' used.  Please see the Scramble
-documentation for details.
+Alter the sequences per slice.  For improved random access values lower than the 10000 default may be considered.
 
-Please note: '-I,-O' are used internally and should not be provided.
+1000 has been tested with older versions of HTSlib and found to be a good trade off, comparible speed 1% increase in file size.
```
[the remark this answers is not in the document]
Comment (keiranmraine) [reply scripts/ds-cgpmap.pl:208]: ? 10000 is the default for htslib, but has an impact on downstream tools.  I've indicated >1000 (which is probably better than +1000) as 1k is a safe trade off that we have tested.

File: cwls/cgpmap-bamBaiOut.cwl [hunk 2]
```diff
@@ -76,14 +76,52 @@ inputs:
       position: 7
       separate: true
 
+  threads:
+    type: int?
+    doc: "Number of CPUs to use where possible"
```
[the remark this answers is not in the document]
Comment (keiranmraine) [reply cwls/cgpmap-bamBaiOut.cwl:81]: I don't like coding around bugs.  Not sure why this stops us releasing.

File: scripts/ds-cgpmap.pl [hunk 3]
```diff
@@ -110,16 +123,23 @@ =head1 SYNOPSIS
...
-    -scramble    -sc  Single quoted string of parameters to pass to Scramble when '-c' used
-                      - '-I,-O' are used internally and should not be provided
-    -bwa         -b     Single quoted string of additional parameters to pass to BWA
-                         - '-t,-p,-R' are used internally and should not be provided
-    -groupinfo   -g   Readgroup metadata file for FASTQ inputs, values are not validated (yaml).
-    -threads     -t   Set the number of cpu/cores available [default all].
-    -outdir      -o   Set the output folder [$HOME]
-    -qc          -q   Apply mismatch QC to reads following duplicate marking
-    -qcf         -f   Mismatch fraction to set as max before failing a read [0.05]
+    -threads     -t    Number of threads to use. [1]
```
[the remark this answers is not in the document]
Comment (keiranmraine) [reply scripts/ds-cgpmap.pl:126]: Good spot

File: cwls/cgpmap-bamBaiOut.cwl (hunk 2 above)
[the remark this answers is not in the document]
Comment (keiranmraine) [reply cwls/cgpmap-bamBaiOut.cwl]: On reflection decided that updating doc and minor change in perl script were worthwhile to cover all bases.

All CLW changed:

```cwl
-    doc: "Number of CPUs to use where possible"
+    doc: "Number of CPUs to use where possible - 0/null/not defined to query host for max"
```

All json changes:

```json
-  "threads": null,
+  "threads": 0,
```

Script change:

```perl
-printf $FH "CPU=%d\n", $opts{'t'} if(defined $opts{'t'});
+printf $FH "CPU=%d\n", $opts{'t'} if(defined $opts{'t'} && $opts{'t'} != 0);
```

**Review by byb121 — APPROVED** [review]

**Pull request merged**